plugins/inputs/aliyuncms/aliyuncms.go

```diff
@@ -326,71 +380,73 @@ func (s *AliyunCMS) updateWindow(relativeTo time.Time) {
 
 // Gather given metric and emit error
 func (s *AliyunCMS) gatherMetric(acc telegraf.Accumulator, metricName string, metric *Metric) error {
-	req := cms.CreateDescribeMetricListRequest()
-	req.Period = strconv.FormatInt(int64(time.Duration(s.Period).Seconds()), 10)
-	req.MetricName = metricName
-	req.Length = "10000"
-	req.Namespace = s.Project
-	req.EndTime = strconv.FormatInt(s.windowEnd.Unix()*1000, 10)
-	req.StartTime = strconv.FormatInt(s.windowStart.Unix()*1000, 10)
-	req.Dimensions = metric.requestDimensionsStr
-
-	for more := true; more; {
-		resp, err := s.client.DescribeMetricList(req)
-		if err != nil {
-			return errors.Errorf("failed to query metricName list: %v", err)
-		} else if resp.Code != "200" {
-			s.Log.Errorf("failed to query metricName list: %v", resp.Message)
-			break
-		}
-
-		var datapoints []map[string]interface{}
-		if err = json.Unmarshal([]byte(resp.Datapoints), &datapoints); err != nil {
-			return errors.Errorf("failed to decode response datapoints: %v", err)
-		}
+	for _, region := range s.Regions {
+		req := cms.CreateDescribeMetricListRequest()
+		req.Period = strconv.FormatInt(int64(time.Duration(s.Period).Seconds()), 10)
+		req.MetricName = metricName
+		req.Length = "10000"
+		req.Namespace = s.Project
+		req.EndTime = strconv.FormatInt(s.windowEnd.Unix()*1000, 10)
+		req.StartTime = strconv.FormatInt(s.windowStart.Unix()*1000, 10)
+		req.Dimensions = metric.requestDimensionsStr
+		req.RegionId = region
+
+		for more := true; more; {
+			resp, err := s.client.DescribeMetricList(req)
+			if err != nil {
+				return errors.Errorf("failed to query metricName list: %v", err)
+			} else if resp.Code != "200" {
+				s.Log.Errorf("failed to query metricName list: %v", resp.Message)
+				break
+			}
```

Comment:
As you return in the `if` you can simplify to
```suggestion
			if err != nil {
				return errors.Errorf("failed to query metricName list: %v", err)
			} 
			if resp.Code != "200" {
				s.Log.Errorf("failed to query metricName list: %v", resp.Message)
				break
			}
```

Reply:
Agree, fixed

Reply:
Still unchanged. I checked out via git and also this code is unchanged.

Reply:
Forgot to push changes, now should be fixed